test/spec.index.js

```diff
@@ -927,6 +940,221 @@ describe('Template Mixins', function () {
+                middleware = mixins({
+                    'field-name': {}
+                });
+                middleware(req, res, next);
+                res.locals['checkbox-group']().call(res.locals, 'field-name');
+                render.should.have.been.calledWithExactly(sinon.match({
+                    hint: null
+                }));
+            });
+
+            describe('multiple options selected', function () {
+                beforeEach(function () {
+                    middleware = mixins({
+                        'field-name': {
+                            options: [{
+                                label: 'Foo',
+                                value: 'foo'
+                            }, {
+                                label: 'Bar',
+                                value: 'bar'
+                            }, {
+                                label: 'Baz',
+                                value: 'baz'
+                            }]
+                        }
+                    });
+                });
+
+                it('sets previously selected options of a group to true', function () {
+                    res.locals.values = {
+                        'field-name': ['foo', 'bar']
+                    };
+                    middleware(req, res, next);
+                    res.locals['checkbox-group']().call(res.locals, 'field-name');
+                    var options = render.args[0][0].options;
+                    _.pluck(options.filter(function (option) {
+                        return option.selected;
+                    }), 'value').should.be.eql(['foo', 'bar']);
+                });
+
+                it('sets previously selected options of a group to true', function () {
+                    res.locals.values = {
+                        'field-name': ['foo', 'bar', 'baz']
+                    };
+                    middleware(req, res, next);
+                    res.locals['checkbox-group']().call(res.locals, 'field-name');
+                    var options = render.args[0][0].options;
+                    _.pluck(options.filter(function (option) {
+                        return option.selected;
+                    }), 'value').should.be.eql(['foo', 'bar', 'baz']);
+                });
+
+                it('sets previously selected options of a group to true', function () {
+                    res.locals.values = {
+                        'field-name': ['foo', 'baz']
+                    };
+                    middleware(req, res, next);
+                    res.locals['checkbox-group']().call(res.locals, 'field-name');
+                    var options = render.args[0][0].options;
+                    _.pluck(options.filter(function (option) {
+                        return option.selected;
+                    }), 'value').should.be.eql(['foo', 'baz']);
+                });
+
+                it('sets previously selected options of a group to true', function () {
```

Comment:
it would be quite a complicated test if we try and cover all in one. What do you suggest for the descriptions?